### PR TITLE
blockio: add blockio_reload option

### PR DESCRIPTION
**Pull request by askervin**
Reconfiguring blockio is needed in the following scenarios.
- Block devices (/dev/*) change, for example when a CSI driver adds a new block device for a pod. Reconfiguration finds current devices and enables applying appropriate blockio parameters for them.
- Blockio class configuration file has been updated in the system. Reconfiguration applies changes without restarting or reconfiguring crio.

Signed-off-by: Antti Kervinen <antti.kervinen@intel.com>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR enables configuring dynamically created block devices. For example, without this patch block devices created by CSI drivers cannot be controlled.

#### Which issue(s) this PR fixes:

The need to rescan block devices was raised by @d-honeybadger [in a discussion](https://github.com/kubernetes/kubernetes/issues/92287#issuecomment-1225647615) on similar blockio handling in containerd.

#### Does this PR introduce a user-facing change?

```release-note
New --blockio-reconfigure flag triggers reloading --blockio-config-file and rescanning block devices in the system every time before applying blockio parameters to a container. This is needed to control blockio parameters of dynamically created block devices.
```

### CODE REVIEW

File: completions/zsh/_crio [hunk 1]
```diff
@@ -28,6 +28,7 @@ it later with **--config**. Global options will modify the output.'
         '--big-files-temporary-dir'
         '--bind-mount-prefix'
         '--blockio-config-file'
+        '--blockio-reconfigure'
```
Comment (haircommander) [inline completions/zsh/_crio:31]: can we use `blockio-reload`? I think it conforms more to what we call this situation internally

**Comment by haircommander** [comment]
/ok-to-test

thanks @askervin LGTM besides one naming item

File: completions/zsh/_crio (hunk 1 above)
Comment (askervin) [reply completions/zsh/_crio]: Thanks @haircommander! Fixed. I think it's better now.

File: pkg/config/template.go [hunk 2]
```diff
@@ -847,6 +852,12 @@ const templateStringCrioRuntimeBlockIOConfigFile = `# Path to the blockio class
 
 `
 
+const templateStringCrioRuntimeBlockIOReload = `# Reload blockio-config-file and rescan blockio devices in the system before applying
+# blockio parameters.
+{{ $.Comment }}blockio_reload = "{{ .BlockIOReload }}"
```
Comment (klihub) [inline pkg/config/template.go:857]: I think you need to ditch the quotes here (`blockio_reload` is a boolean).

Comment (askervin) [reply pkg/config/template.go]: Thanks, @klihub! Fixed.

**Comment by TomSweeneyRedHat** [comment]
LGTM
once the tests are hip

**Comment by haircommander** [comment]
/approve

@saschagrunert PTAL

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by klihub** [comment]
@askervin Could you rebase this on latest main HEAD ?

**Comment by askervin** [comment]
Thanks @klihub! Rebased and tested.

**Comment by m-yosefpor** [comment]
any updates on this PR? can we help to push it forward? this issue is affecting us in production.

**Comment by sohankunkerkar** [comment]
CI test is failing: https://github.com/cri-o/cri-o/actions/runs/5951611023/job/16141913786?pr=6254#step:5:18

**Pull request merged**